front/pages/w/[wId]/builder/data-sources/[name]/index.tsx

```diff
@@ -940,18 +949,39 @@ function ManagedDataSourceView({
     addDataButtonLabel,
     displaySettingsButton,
     guideLink,
+    onPostPermissionsUpdate,
   } = getRenderingConfigForConnectorProvider(connectorProvider);
 
+  const [
+    postPermissionsUpdateDialogIsOpen,
+    setPostPermissionsUpdateDialogIsOpen,
+  ] = useState(false);
+
   return (
     <>
+      {onPostPermissionsUpdate && (
+        <Dialog
+          isOpen={postPermissionsUpdateDialogIsOpen}
+          onCancel={() => setPostPermissionsUpdateDialogIsOpen(false)}
+          onValidate={() => {
+            setPostPermissionsUpdateDialogIsOpen(false);
+          }}
+          title="Permissions updated"
+        >
+          {onPostPermissionsUpdate()}
+        </Dialog>
+      )}
       <DataSourceDetailsModal
         dataSource={dataSource}
         visible={showDataSourceDetailsModal}
         onClose={() => {
           setShowDataSourceDetailsModal(false);
         }}
         onClick={() => {
-          void handleUpdatePermissions();
+          void handleUpdatePermissions().then(() => {
```

Comment:
nit: I think we had something saying from now on we need to catch?

Reply:
In client-side code too ? I'm not sure what I would be doing with it (console.error is not actionable for the user)

Reply:
yeah true